src/Microsoft.Health.Fhir.Api/Extensions/HttpRequestExtensions.cs

```diff
@@ -24,5 +27,36 @@ public static bool IsFhirRequest(this HttpRequest request)
             return !request.Path.StartsWithSegments(KnownRoutes.HealthCheck, StringComparison.InvariantCultureIgnoreCase) &&
                    !request.Path.StartsWithSegments(KnownRoutes.CustomError, StringComparison.InvariantCultureIgnoreCase);
         }
+
+        /// <summary>
+        /// Return a group of search parameters from a query string.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        public static IReadOnlyList<Tuple<string, string>> GetQueriesForSearch(this HttpRequest request)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            if (request.QueryString.HasValue)
+            {
+                NameValueCollection queryStringParameters =
+                    HttpUtility.ParseQueryString(
+                            request.QueryString.Value);
+
+                List<Tuple<string, string>> queries = new List<Tuple<string, string>>();
```

Comment:
Just curious here, is there a reason to not just use a Dictionary<string> here instead of List<Tuple<string, string>>?

Reply:
That's a great question.
And there is a reason: If customer uses the query string "?name=FOO&name=BAR", then the final result should have (name, FOO) and (name, BAR), because that's an AND. 

If we adopt the Dictionary, then it would not be possible to handle this case, because the dictionary would not accept two keys with the same value.